File: src/main/kotlin/tech/diggle/apps/accomodation/elections/Election.java
package tech.diggle.apps.accomodation.elections;


import org.springframework.format.annotation.DateTimeFormat;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.util.Date;

@Entity
@TableGenerator(name = "elections")
public class Election {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @NotNull
    @Column(name = "id")
    private long id;

    @NotNull
    @Column(name = "electionType")
    private String electionType;

    @NotNull
    @Column(name = "electionDate")
    @DateTimeFormat(pattern = "yyyy-MM-dd")
    private Date electionDate;

    public Election(@NotNull String electionType, @NotNull Date electionDate) {
        this.electionType = electionType;
        this.electionDate = electionDate;
    }

    public Election() {

    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getElectionType() {
        return electionType;
    }

    public void setElectionType(String electionType) {
        this.electionType = electionType;
    }

    public Date getElectionDate() {
        return electionDate;
    }

    public void setElectionDate(Date electionDate) {
        this.electionDate = electionDate;
    }
}

File: src/main/kotlin/tech/diggle/apps/accomodation/election/Election.kt
package tech.diggle.apps.accomodation.election

import org.springframework.format.annotation.DateTimeFormat
import tech.diggle.apps.accomodation.candidate.Candidate
import java.util.*
import javax.persistence.*
import javax.validation.constraints.NotNull

@Entity
class Election {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    val id: Long = 0

    @NotNull
    @Column(name = "type")
    var type: ElectionType? = null

    @NotNull
    @Column(name = "date")
    @DateTimeFormat(pattern = "yyyy-MM-dd")
    var date: Date? = null

    @NotNull
    @DateTimeFormat(pattern = "yyyy-MM-dd'T'hh:mm:ss.SSSZ ")
    var startDate: Date? = null

    @NotNull
    @DateTimeFormat(pattern = "yyyy-MM-dd'T'hh:mm:ss.SSSZ ")
    var endDate: Date? = null

    var status: ElectionStatus? = null

    @OneToMany
    var candidates: List<Candidate> = emptyList()
}

